pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -426,8 +430,17 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation, errorContext *er
 			Dependencies: flow.NewTaskIDs(destroyInfrastructure),
 		})
 		destroyExternalDomainDNSRecord = g.Add(flow.Task{
-			Name:         "Destroying external domain DNS record",
-			Fn:           flow.TaskFn(botanist.DestroyExternalDomainDNSRecord).DoIf(dnsEnabled),
+			Name: "Destroying external domain DNS record",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return botanist.ExternalDNS().Destroy(ctx)
+			}),
+			Dependencies: flow.NewTaskIDs(syncPointCleaned),
+		})
+		destroyInternalDomainDNSRecord = g.Add(flow.Task{
+			Name: "Destroying internal domain",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
```

Comment:
You can use directly the target fn (without the wrapper one) , right?

```suggestion
			Fn:           botanist.InternalDNS().Destroy,
```